File: web/src/api/sysRole.ts
import { http,resData } from "@/utils/http";


// @Tags SysRole
// @Summary 创建SysRole
export const createSysRole = (data?: object) => {
  return http.request<resData>("post", "/sysRole/createSysRole", { data });
};

// @Tags SysRole
// @Summary 删除SysRole
export const deleteSysRole = (data?: object) => {
  return http.request<resData>("post", "/sysRole/deleteSysRole", { data });
};


// @Tags SysRole
// @Summary 删除SysRole
export const deleteSysRoleByIds = (data?: object) => {
  return http.request<resData>("post", "/sysRole/deleteSysRoleByIds", { data });
};


// @Tags SysRole
// @Summary 更新SysRole
export const updateSysRole = (data?: object) => {
  return http.request<resData>("post", "/sysRole/updateSysRole", { data });
};


// @Tags SysRole
// @Summary 用id查询SysRole
export const findSysRole =  (params?: object) => {
  return http.request<resData>("get", "/sysRole/findSysRole", { params });
};


// @Tags SysRole
// @Summary 分页获取SysRole列表
export const getSysRoleList =  (params?: object) => {
  return http.request<resData>("get", "/sysRole/getSysRoleList", { params });
};




// @Tags SysRole
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
  return http.request<resData>("post", "/sysRole/quickEdit", { data });
};


// @Tags SysRole
// @Summary 导出excelSysRole列表
export const excelList =(data?: object) => {
  return http.request<resData>("get", "/sysRole/excelList", { data });
};

File: server/router/business/sysRole.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type SysRoleRouter struct {
}

// InitSysRoleRouter 初始化 SysRole 路由信息
func (s *SysRoleRouter) InitSysRoleRouter(Router *gin.RouterGroup) {
	router := Router.Group("sysRole").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("sysRole")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.SysRoleApi
	{
		router.POST("createSysRole", apiV1.CreateSysRole)   // 新建SysRole
		router.POST("deleteSysRole", apiV1.DeleteSysRole) // 删除SysRole
		router.POST("deleteSysRoleByIds", apiV1.DeleteSysRoleByIds) // 批量删除SysRole
		router.POST("updateSysRole", apiV1.UpdateSysRole)    // 更新SysRole
	    router.POST("quickEdit", apiV1.QuickEdit)  // 快速编辑	
	}
	{
		routerNoRecord.GET("findSysRole", apiV1.FindSysRole)        // 根据ID获取SysRole
		routerNoRecord.GET("getSysRoleList", apiV1.GetSysRoleList)  // 获取SysRole列表
        routerNoRecord.GET("excelList", apiV1.ExcelList)  // 分页导出excel SysRole列表
	}
}
